pkg/services/authn/grpcutils/grpc_authenticator.go

```diff
@@ -117,36 +117,42 @@ func (f *AuthenticatorWithFallback) Authenticate(ctx context.Context) (context.C
 	if err != nil {
 		// In case of error, fallback to the legacy authenticator
 		newCtx, err = f.fallback.Authenticate(ctx)
-		f.metrics.fallbackCounter.WithLabelValues(fmt.Sprintf("%t", err == nil)).Inc()
 		span.SetAttributes(attribute.Bool("fallback_used", true))
+		if err != nil {
```

Comment:
I find it a bit hard to read, what do you think of
```go
func (f *AuthenticatorWithFallback) Authenticate(ctx context.Context) (context.Context, error) {
	ctx, span := f.tracer.Start(ctx, "grpcutils.AuthenticatorWithFallback.Authenticate")
	defer span.End()
	span.SetAttributes(attribute.Bool("fallback_used", false))
	// Try to authenticate with the new authenticator first
	newCtx, err := f.authenticator.Authenticate(ctx)
	if err == nil {
		// fallback not used, authentication successful
		f.metrics.requestsTotal.WithLabelValues("false", "true").Inc()
		return newCtx, nil
	}

	// In case of error, fallback to the legacy authenticator
	span.SetAttributes(attribute.Bool("fallback_used", true))
	newCtx, err = f.fallback.Authenticate(ctx)	
	if err == nil {
		// fallback used, authentication successful
		newCtx = context.WithValue(newCtx, contextFallbackKey{}, true)
		f.metrics.requestsTotal.WithLabelValues("true", "true").Inc()
	} else {
		// fallback used, authentication failed
		f.metrics.requestsTotal.WithLabelValues("true", "false").Inc()
	}
	return newCtx, err
}
```

Reply:
👍